iceoryx_posh/source/mepoo/chunk_header.cpp

```diff
@@ -90,20 +90,30 @@ ChunkHeader::ChunkHeader(const uint32_t chunkSize, const ChunkSettings& chunkSet
         //     -> the back-offset can be put adjacent to to the Topic since the smaller alignment always fits
         auto backOffsetAddress = alignedUserPayloadAddress - sizeof(UserPayloadOffset_t);
         auto backOffset = reinterpret_cast<UserPayloadOffset_t*>(backOffsetAddress);
-        *backOffset = userPayloadOffset;
+        *backOffset = m_userPayloadOffset;
     }
 
     cxx::Ensures(overflowSafeUsedSizeOfChunk() <= chunkSize
                  && "Used size of chunk would exceed the actual chunk size!");
 }
 
-void* ChunkHeader::userPayload() const noexcept
+uint8_t ChunkHeader::chunkHeaderVersion() const noexcept
+{
+    return m_chunkHeaderVersion;
+}
+
+void* ChunkHeader::userPayload() noexcept
```

Comment:
Shouldn't this be templated?

Reply:
Could be, but it's only useful for the untyped API. If we change this now, it's quite a lot of work

Reply:
I know what you mean. What can happen worst case? For me it's important that we ensure, that the typed API Users always has full type-safety.

Reply:
They have. They get a `Sample<T, H>` with methods to get the `ChunkHeader` and the user-header `H`. Derefererencing the sample accesses the payload in a typesafe way.